packages/@react-spectrum/combobox/src/ComboBox.tsx

```diff
@@ -220,9 +261,17 @@ const ComboBoxInput = React.forwardRef(function ComboBoxInput(props: ComboBoxInp
               'spectrum-InputGroup-input'
             )
           }
+          validationIconClassName={
+            classNames(
+              styles,
+              'spectrum-InputGroup-input-validationIcon'
+            )
+          }
           isDisabled={isDisabled}
           isQuiet={isQuiet}
-          validationState={validationState} />
+          validationState={validationState}
+          isLoading={isLoading}
+          loadingIcon={loading} />
```

Comment:
Should this be more like `spectrum-Textfield--loadable`? i.e. should we always reserve space for the spinner if it could show up so that when it does it doesn't shift the contents of the textfield? In that case, we could just look for the presence of the `loadingIcon` prop?

Also IMO we could rename `loadingIcon` to e.g. `loadingIndicator` since it's not really an icon?

Reply:
That sounds fine, lemme adjust

Reply:
Looks a little funny for the tray search field but stops the shifting that you mentioned previously:
![image](https://user-images.githubusercontent.com/9661127/104773122-61bd3e80-5729-11eb-887e-83686d920b43.png)

Reply:
Maybe we should swap the location of the clear button and the loading indicator, so the loading indicator is to the left of the clear button? And also reserve space for both of them at all times? We should probably also only show the loading indicator if there is text in the input so that it doesn't show up without the clear button...

Reply:
Sure I can go ahead and do that. As for showing the loading indicator only if there is text in the input, are we concerned about if the field is empty and the first load takes a while? The user would only see a empty tray and no indicator of the current loading state.

Perhaps I could have the clear button appear at all times? Open to ideas

Reply:
Another issue that comes up is the positioning of the validation icons. By flipping the loading indicator position and the clear button position, the clear button overlaps the validation icon since they are now placed in the same position. Changing the positioning of the validation icon is a bit tricky cuz it is within the textfield css

Reply:
For now I've made it so the clear button appears at all times and have flipped the clear button's position with the loading indicator/validation icon (leveraged the `validationIconClassName` prop I forgot I added) position. Also made it so the clear button and validation icon positions for non-async comboboxes were flipped for consistency. Lemme know what you think